src/main/java/minijava/ir/optimize/ConstantControlFlowOptimizer.java

```diff
@@ -14,7 +18,7 @@
  * <p>The {@link Proj} node that is no longer relevant is replaced with a {@link Bad} node. A
  * subsequent run of an {@link Optimizer} that removes such nodes is required.
  */
```

Comment:
I hoped you would implement this in a separate class. I guess the name of this class suggested that this actually belongs here ;)
If you really don't want to put these optimizations in separate classes (I would prefer that though), then please adjust the class documentation.